### PR TITLE
added the route for recepie page

### LINKED ISSUE #43: configure a route for detailed recipe
- Identify what information is needed to display details for user selected recipe (ex: slug, id..)
- Configure a route for detailed recipe view
- When navigated to this route (ex: `/recipe/<id | slug>` ) being able to see feedback that user is trying to view a particular recipe would be nice (ex: `Displaying recipe details for : id | slug`

### PR DESCRIPTION
**Pull request by singhpriya147**
added the route for individual recepie page which is renderd when user click on meals card in homepage/landing page

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -2,12 +2,14 @@ import { Route, Routes } from "react-router-dom";
...
 import Layout from "./components/layout/Layout";
 import Home from "./pages/home/Home";
+import DetailRecepie from "./pages/home/recepie";
 
 export default function App() {
   return (
     <Layout>
       <Routes>
         <Route path="/" element={<Home />} />
+        <Route path="/recepie" element={<DetailRecepie />} />
```
Comment (diaasaur) [inline src/App.tsx]: two things:
- i think a small typo slipped in and persisted throughout the changes ,`recepie` should be `recipe` 
- why no url param like `id` ? ( based on what is needed to get more details for a recipe) 

please refer: [url params](https://v5.reactrouter.com/web/example/url-params)

File: src/pages/home/recepie.tsx [hunk 2]
```diff
@@ -0,0 +1,4 @@
+function DisplayRecepie() {
+  return <div>This is the recepie page</div>;
```
Comment (diaasaur) [inline src/pages/home/recepie.tsx]: this works! but I think a variant of this message that includes the id (pulled in from url param) will be better :D - that way developer will get an instant feedback from this skeleton route.

an example would be like: `Displaying recipe details for : id` as mentioned in the task

**Comment by diaasaur** [comment]
and one more thing, since detailed recipe view is it's own page, it's better to move it out of `pages/home` to a new folder under pages, maybe like: `pages/recipe-details` ? or `pages/detailed-recipe`? whichever works! 

we are _loosely_ following the folder structure video adham sent in the group chat :3

File: src/App.tsx (hunk 1 above)
Comment (singhpriya147) [reply src/App.tsx]: yeah ... i am really sorry for the typo.  Though i added new route with params .  and i am using some dummy api to mimic the "fetching by id" functionality . you guys should check it once more by changing the url by "/recipe/id" where id can be 1,2,3,4 etc

Comment (DikshantAgarwal) [reply src/App.tsx]: You can add the code spell checker extension on your visual studio code.
image

File: src/pages/Recipe.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+interface Post {
+  userId: number;
+  id: number;
+  title: string;
+  body: string;
+}
+
+const retrievePost = async (postId: number): Promise<Post> => {
+  const response = await axios.get<Post>(
+    `https://jsonplaceholder.typicode.com/posts/${postId}`,
```
Comment (DikshantAgarwal) [inline src/pages/Recipe.tsx:15]: Can we create a separate folder altogether for the Api-related stuff?
  i.e  api
         ---reciepeType.ts
         ---getRecipe.ts( contain export functions which will call API) similar to function retrievePost

We can discuss more on this.

File: src/App.tsx (hunk 1 above)
Comment (singhpriya147) [reply src/App.tsx]: oh... thanks.... btw that's really great

**Comment by diaasaur** [comment]
this looks good and gave me an overview of how react-query works! i'm not really sure if we should push jsonplaceholder into develop though, but isn't that what develop is for? a testing playground 🧐- other than that, I think we will move out api + react-query into an appropriate folder(s) as discussed, in future PRs.

approving this PR, meanwhile - do let me know about the dummy api, which can be commented out if it's a problem 🫡

**Review by diaasaur — DISMISSED** [review]

**Comment by DikshantAgarwal** [comment]
If we are looking at it as just react-query setup then I think it's fine.  @singhpriya147  you can put a comment before the dummy API i.e 'this is just a dummy API for setup, will be removed later"  so that everyone can get the reference. 

Approved.

**Review by DikshantAgarwal — DISMISSED** [review]

**Review by diaasaur — APPROVED** [review]

**Review by Sumitgangwar58 — APPROVED** [review]

**Pull request merged**